// patternfly-react/dist/js/components/VerticalNav/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VerticalNavBadge = exports.VerticalNavIconBar = exports.VerticalNavBrand = exports.VerticalNavTertiaryItem = exports.VerticalNavSecondaryItem = exports.VerticalNavItem = exports.VerticalNavMasthead = exports.VerticalNav = undefined;

var _VerticalNav = require('./VerticalNav');

var _VerticalNav2 = _interopRequireDefault(_VerticalNav);

var _VerticalNavMasthead = require('./VerticalNavMasthead');

var _VerticalNavMasthead2 = _interopRequireDefault(_VerticalNavMasthead);

var _VerticalNavItem = require('./VerticalNavItem');

var _VerticalNavItem2 = _interopRequireDefault(_VerticalNavItem);

var _VerticalNavSecondaryItem = require('./VerticalNavSecondaryItem');

var _VerticalNavSecondaryItem2 = _interopRequireDefault(_VerticalNavSecondaryItem);

var _VerticalNavTertiaryItem = require('./VerticalNavTertiaryItem');

var _VerticalNavTertiaryItem2 = _interopRequireDefault(_VerticalNavTertiaryItem);

var _VerticalNavBrand = require('./VerticalNavBrand');

var _VerticalNavBrand2 = _interopRequireDefault(_VerticalNavBrand);

var _VerticalNavIconBar = require('./VerticalNavIconBar');

var _VerticalNavIconBar2 = _interopRequireDefault(_VerticalNavIconBar);

var _VerticalNavBadge = require('./VerticalNavBadge');

var _VerticalNavBadge2 = _interopRequireDefault(_VerticalNavBadge);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

_VerticalNav2.default.Masthead = _VerticalNavMasthead2.default;
_VerticalNav2.default.Item = _VerticalNavItem2.default;
_VerticalNav2.default.SecondaryItem = _VerticalNavSecondaryItem2.default;
_VerticalNav2.default.TertiaryItem = _VerticalNavTertiaryItem2.default;
_VerticalNav2.default.Brand = _VerticalNavBrand2.default;
_VerticalNav2.default.IconBar = _VerticalNavIconBar2.default;
_VerticalNav2.default.Badge = _VerticalNavBadge2.default;

exports.VerticalNav = _VerticalNav2.default;
exports.VerticalNavMasthead = _VerticalNavMasthead2.default;
exports.VerticalNavItem = _VerticalNavItem2.default;
exports.VerticalNavSecondaryItem = _VerticalNavSecondaryItem2.default;
exports.VerticalNavTertiaryItem = _VerticalNavTertiaryItem2.default;
exports.VerticalNavBrand = _VerticalNavBrand2.default;
exports.VerticalNavIconBar = _VerticalNavIconBar2.default;
exports.VerticalNavBadge = _VerticalNavBadge2.default;
